File: core/tmp/Python/stack/clumsy_factorial.py
def clumsy(N: int) -> int:
    num = list()
    num.append(N)
    flag = 0
    for i in range(N - 1, 0, -1):
        if flag == 0:
            num.append(i * num.pop())
        elif flag == 1:
            num.append(int(num.pop() / i))
        elif flag == 2:
            num.append(i)
        elif flag == 3:
            num.append(-i)
        flag = (flag + 1) % 4

    total = 0
    print(num)
    while num:
        total += num.pop()
    return total

if __name__ == "__main__":
    print(clumsy(10))

File: core/tmp/Java/src/com/dbc/algorithm/Stack/ClumsyFactorial.java
package com.dbc.algorithm.Stack;

import java.util.Stack;

public class ClumsyFactorial {
    public int clumsy(int N) {
        Stack<Integer> stack = new Stack<>();
        stack.push(N);
        int flag = 0;
        for (int i = N - 1; i > 0; i--) {
            if (flag % 4 == 0) {
                stack.push(stack.pop() * i);
            } else if (flag % 4 == 1) {
                stack.push(stack.pop() / i);
            } else if (flag % 4 == 2) {
                stack.push(i);
            } else if (flag % 4 == 3) {
                stack.push(-i);
            }
            flag++;
        }

        int total = 0;
        while (!stack.isEmpty()) {
            total += stack.pop();
        }
        return total;
    }
}
